src/shims/unix/sync.rs

```diff
@@ -435,7 +354,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         let this = self.eval_context_mut();
 
         let kind = mutex_get_kind(this, mutex_op)?;
-        let id = mutex_get_or_create_id(this, mutex_op)?;
+        let id = this.mutex_get_or_create_id(mutex_op, 4)?;
```

Comment:
Please define some `const` for this 4. It used to be in one place only but your refactor lead to duplication. (Same for the other offsets in this file and the Windows file.)